sdks/python/apache_beam/ml/transforms/utils.py

```diff
@@ -28,8 +30,13 @@ class ArtifactsFetcher():
   to the TFTProcessHandlers in MLTransform.
   """
   def __init__(self, artifact_location):
-    self.artifact_location = artifact_location
-    self.transform_output = tft.TFTransformOutput(self.artifact_location)
+    files = os.listdir(artifact_location)
+    files.remove(base._ATTRIBUTE_FILE_NAME)
+    if len(files) > 1:
+      raise NotImplementedError(
+          'Multiple files in artifact location not supported yet.')
```

Comment:
User wouldn't know since we create a  random directory inside artifact location. I added more info on the NotImplementedError. One idea is to integrate ArtifactsFetcher into MLTransform but can be addressed later

Reply:
I'm not quite sure I follow this error message. Is this saying you can't have something like:

`MLTransform().with_transform(tft_operation).with_transform(embeddings)`? I thought that's what the logic in base.py around `append_transform` was supposed to handle. Or is this saying you can't use MLTransform and vanilla TFT in the same artifact location? Feel free to take conversation offline if helpful

Reply:
This class is for the users to manually look up artifacts. Since we introduce creating sub-directories in a user specified artifact location, if the user has 2 jobs with `MLTransform().with_transform(tft_operation).with_transform(embeddings)` that have the same artifact location, there would be two sub-directories created by `MLTransform`.  By using ArtifactsFetcher, we don't know which artifact location to read and we will need a better solution for that.

Reply:
Actually, we should throw an error or warn the user if they use the same artifact location. TFT doesn't allow it right now and we should follow the same pattern.

Reply:
Got it, missed the purpose of this function - I think this is now resolved via `save_attributes` changes so we can probably resolve this thread